pkgs/applications/editors/vscode/extensions/rescript/rescript-editor-analysis.nix

```diff
@@ -1,21 +1,31 @@
-{ lib, stdenv, fetchFromGitHub, bash, ocaml }:
+{ lib, stdenv, fetchFromGitHub, bash, ocaml, dune_3, extensionVersion }:
 
 stdenv.mkDerivation {
   pname = "rescript-editor-analysis";
-  version = "1.1.3";
+  version = extensionVersion;
 
   src = fetchFromGitHub {
     owner = "rescript-lang";
     repo = "rescript-vscode";
-    rev = "8d0412a72307b220b7f5774e2612760a2d429059";
-    sha256 = "rHQtfuIiEWlSPuZvNpEafsvlXCj2Uv1YRR1IfvKfC2s=";
+    rev = extensionVersion;
+    sha256 = "sha256-28Z6pB+Vwwtw2MHNlDUCUhuD93oOcVyzMfGdMPLY/s8=";
   };
 
-  nativeBuildInputs = [ ocaml ];
+  nativeBuildInputs = [ ocaml dune_3 ];
 
   postPatch = ''
+    # Skip (sub)steps which involve installing node modules from npm, because:
+    # 1. None of these steps are critical to building the binary output of this derivation. These steps build examples as test cases.
+    # 2. These steps have failed to build in every which way:
+    #    These steps involve `npm install` which needs internet connection, which is not allowed in nix-build.
+    #    None of the usual alternatives succeeded with the offline approach of using package-lock.json.
+    #    The alternatives are either are not compatible with npm's v2 of package-lock.json
+    #    (node2nix, npmlock2nix, https://github.com/nix-community/npmlock2nix/issues/153),
+    #    which is what the upstream uses,
+    #    or they (nix-npm-buildpackage) fail with some other semi-related errors (https://github.com/nix-community/npmlock2nix/issues/45)
```

Comment:
I would recommend to move this out of the string to easily change it in the future without triggering a rebuild

Reply:
Nice catch! Never thought about this.